### PR TITLE
doc: update howto guide for generated libraries

### PR DESCRIPTION
**Pull request by coryan**
The `check-api` build immediately starts testing new GA libraries, we need to create the API baseline as we create the new library.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10516)

### CODE REVIEW

File: doc/contributor/howto-guide-adding-generated-libraries.md [hunk 1]
```diff
@@ -253,6 +253,17 @@ Announce the new library in the CHANGELOG for the next release.
...
 
+## Add the API baseline
+
+For new GA libraries you need to create the API baseline.
+
+```
+ci/cloudbuild/build.sh -t check-api-pr
+git add ci/abi-dumps/google_cloud_cpp_${library}.expected.abi.dump.gz
+git commit -m"Add API baseline" ci/abi-dumps/google_cloud_cpp_${library}.expected.abi.dump.gz
+git checkout -- ci/abi-dumps/
```
Comment (devbww) [inline doc/contributor/howto-guide-adding-generated-libraries.md:264]: Why is this needed?  Isn't the working tree already in the right shape?

Comment (coryan) [reply doc/contributor/howto-guide-adding-generated-libraries.md:264]: The working tree contains (trivial) changes for all API dumps.  And we are going to commit the full directory later.  I would rather avoid the churn.

Comment (devbww) [reply doc/contributor/howto-guide-adding-generated-libraries.md:264]: I see.  Might not `git restore ci/abi-dumps` be clearer?

**Review by devbww — APPROVED** [review]

**Comment by coryan** [comment]
Done

**Review by devbww — APPROVED** [review]
